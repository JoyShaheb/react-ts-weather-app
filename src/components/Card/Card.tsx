import { FC } from "react";
import { Typography } from "@mui/material";
import { Stack } from "@mui/system";

interface iCard {
  icon: JSX.Element | JSX.Element[];
  label: string;
  data: string | undefined | number;
  unit?: string;
}

const Card: FC<iCard> = ({ icon, label, data, unit }) => {
  return (
    <Stack
      sx={{
        border: "2px solid gray",
        borderRadius: "3px",
        padding: "10px",
        height: "125px",
      }}
      gap={0.5}
      direction="column"
      justifyContent="center"
      alignItems="center"
      className="Card"
    >
      {icon}
      <Typography>
        {data} {unit}
      </Typography>
      <Typography variant="body2" color="text.secondary">
        {label}
      </Typography>
    </Stack>
  );
};

export default Card;
