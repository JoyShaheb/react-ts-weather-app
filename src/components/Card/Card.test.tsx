import { render, screen } from "@testing-library/react";
import Card from "./Card";

describe("Card component", () => {
  const sampleData = {
    icon: <span>ICON</span>,
    label: "Label",
    data: "Data",
    unit: "Unit",
  };
  it("should render the correct icon, label, data, and unit", () => {
    // Arrange
    render(<Card {...sampleData} />);

    // Act
    const iconElement = screen.getByText(/ICON/i);
    const labelElement = screen.getByText(/Label/i);
    const dataElement = screen.getByText(/Data/i);
    const unitElement = screen.getByText(/Unit/i);

    // Assert
    expect(iconElement).toBeInTheDocument();
    expect(labelElement).toBeInTheDocument();
    expect(dataElement).toBeInTheDocument();
    expect(unitElement).toBeInTheDocument();
  });

  it("snapshot test of Card component", () => {
    const { container } = render(<Card {...sampleData} />);
    expect(container).toMatchSnapshot();
  });
});
