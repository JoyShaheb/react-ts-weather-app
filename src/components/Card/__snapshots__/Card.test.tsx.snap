// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component snapshot test of Card component 1`] = `
<div>
  <div
    class="MuiStack-root Card css-1sy57ly-MuiStack-root"
  >
    <span>
      ICON
    </span>
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Data
       
      Unit
    </p>
    <p
      class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
    >
      Label
    </p>
  </div>
</div>
`;
